### ISSUE TITLE
[2] Settlement Price on Price Chart

### ISSUE THREAD

**Issue by fractalnode** [bitshares/bitshares-ui#502]
Settlement Price on Price Chart as an additional layer – just as EMA

**Comment by ai3d-brodie** [comment]
And orders on the chart.  yeaaaaah

**Comment by wmbutler** [comment]
#361

**Comment by wmbutler** [comment]
Can we make this a tradingview layer?

**Comment by startailcoon** [comment]
Is this relevant to the current chart, or are we waiting for trading view?

**Comment by wmbutler** [comment]
I think this was completed.

![screen shot 2018-02-21 at 1 24 42 pm](https://user-images.githubusercontent.com/1254810/36500709-a5502ee4-170a-11e8-8b94-5cbef6f4fbc7.png)

**Comment by wmbutler** [comment]
This should depend on #361

**Comment by clockworkgr** [comment]
Don't think the nodes keep a history of this per se. They only keep a history of the various feeds submitted  (and limited one at that) + the current consensus median feed price.

This cannot be computed client side either because limited API history and different number of operations between feeders means you might end up with missing entries if trying to recreate past consensus client-side.

This will have to be done via recreating past consensus client-side utilising a full or ES node and seems quite resource intensive and difficult to implement.

Unless @oxarbitrage / @abitmore etc. can provide an API for it. Doubtful since it would require storing all the historic price feed changes....a lot of them. But tagging them to confirm before closing as wont-fix

**Comment by abitmore** [comment]
IIRC historical price feeds are stored in es_objects plugin. @oxarbitrage may know better.

**Comment by startailcoon** [comment]
If not mistaken, haven't tested though, es_objects does store feed price. 

Relevant code bits are

https://github.com/bitshares/bitshares-core/blob/cdc2db30c2f06aaddbfda965ee270b99dc24e0aa/libraries/plugins/es_objects/es_objects.cpp#L307

https://github.com/bitshares/bitshares-core/blob/cdc2db30c2f06aaddbfda965ee270b99dc24e0aa/libraries/plugins/es_objects/es_objects.cpp#L256-L257

**Comment by sschiessl-bcp** [comment]
It never occured to me that this might be usefull, but it certainly is!

Let's push it

**Comment by startailcoon** [comment]
I was just going back to this today as I reflected on it again when trading and noticed your response. Making it possible to see feed price indications or historic events will make it possible to see how other markets goes based on the feeds. This would be valuable for the trading on the DEX I believe.

**Comment by bitcrab** [comment]
in my view we need not only to include median feed price to the chart, but also need to include the feed price of each witness, which will greatly help voters to estimate the price feeding quality of each witness.

**Comment by startailcoon** [comment]
I've thought about the same thing as @bitcrab here, that we should make it possible to see the witnesses feed prices visually as well, but this should in such case be in a different place and not on the exchange chart. Historic prices are of interest here I think, but in case we want to do something we can always pull the current ones and display them in a visual chart.

On the exchange chart I'd like to view the same things as what we have on the market depth chart, such as feed price, global settlement and call order limits. However, I could not find anything about how to draw on the trading view chart, it would need some researching.

**Comment by sschiessl-bcp** [comment]
My suggestion: We add view-specific endpoints to the es-wrapper and use them just like a HTTP GET call with json body response.

Thoughts?

**Comment by startailcoon** [comment]
> My suggestion: We add view-specific endpoints to the es-wrapper and use them just like a HTTP GET call with json body response.
> 
> Thoughts?

To populate a feed history graph, sounds about right yes.

**Comment by sschiessl-bcp** [comment]
> > My suggestion: We add view-specific endpoints to the es-wrapper and use them just like a HTTP GET call with json body response.
> > Thoughts?
> 
> To populate a feed history graph, sounds about right yes.

Can you coordinate with explorer api please? @oxarbitrage is probably willing to implement such an endpoint for us?

**Comment by oxarbitrage** [comment]
Yes, thank you for submitting an issue into the explorer. Please submit all you guys think is needed. @Zapata and i will try to split the load. Thanks.

**Comment by Zapata** [comment]
I have started to look at this, please provide your feedback there: bitshares/bitshares-explorer-api#53  ([this comment](https://github.com/bitshares/bitshares-explorer-api/issues/53#issuecomment-517937550) and [this one](https://github.com/bitshares/bitshares-explorer-api/issues/53#issuecomment-518003084)).
